fs/compress.py

```diff
@@ -90,7 +90,8 @@ def write_zip(src_fs,                            # type: FS
             zip_info = zipfile.ZipInfo(zip_name, zip_time)  # type: ignore
 
             try:
-                zip_info.external_attr = info.permissions.mode << 16
+                if info.permissions:
```

Comment:
`info.permissions` could be None according to the typing information. I'm not certain if it could be None in practice.

Perhaps a cast would be more appropriate there?

Reply:
I'm not certain, but what about a filesystem with a concept of ownership but no notion of permissions ? It would then only define the `access` namespace partially. IMO we should not make the assumption that `info.permissions` is never `None`.

(Shouldn't it you use `if info.permissions is not None` just to be explicit ? I had to look to the `fs.permissions` source to check that `Permissions.__bool__` was not implemented)

Reply:
Good point. Should definitely be an explicit check for None.